api/v1alpha3/azurecluster_validation.go

```diff
@@ -26,6 +26,11 @@ import (
 )
 
 const (
+	// can't use: \/"'[]:|<>+=;,?*@&, Can't start with underscore. Can't end with period or hyphen.
```

Comment:
What happens with the DNS entry for the cluster if the cluster name contains a period?

Reply:
the regex actually doesn't allow for . for that reason, I'll update the comment

Reply:
`Can't end with period or hyphen.` -> if the cluster name can't contain a period at all should we move `.` to the list of forbidden symbols and remove it from "end with"?

Reply:
added the `.` and there is also a comment in the following line